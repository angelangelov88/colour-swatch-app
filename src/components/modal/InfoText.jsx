import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'

const InfoText = (props) => {
  let {
    children,
    className,
    color,
  } = props

  const classes = classNames(
    "my-6 py-5 px-6 text-sm font-medium",
    color === "primary" && "text-primary-100 bg-primary-10 shadow-[inset_6px_0px_0px_#4456F6]",
    color === "green" && "text-green-100 bg-green-10 shadow-[inset_6px_0px_0px_#46DBAA]",
    color === "yellow" && "text-yellow-100 bg-yellow-10 shadow-[inset_6px_0px_0px_#F0B618]",
    color === "red" && "text-red-100 bg-red-10 shadow-[inset_6px_0px_0px_#F64A44]",
    className
  )

  return (
    <div className={classes}>
      <p>{ children }</p>
    </div>
  )
}

InfoText.defaultProps = {
  children: "There are a few ways to get started with a new document in Legalesign. Start by telling us what youâ€™d like to do:",
  color: "primary"
}

InfoText.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
  color: PropTypes.oneOf([
    "primary",
    "green",
    "yellow",
    "red",
  ]),

}

export { InfoText }
