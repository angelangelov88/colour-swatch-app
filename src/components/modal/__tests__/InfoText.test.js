import renderer from 'react-test-renderer'
import {InfoText} from "../InfoText"

it('should render the element with red text and red background plus the text stated in children', () => {
  const component = renderer.create(
    <InfoText 
    color="red"
    children="This is a test"
  />
);
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});


it('should render the element in yellow with no text', () => {
  const component = renderer.create(
    <InfoText 
    color="yellow"
    children=""
  />
);
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});


it('should render the element in green with heading with bold text and text', () => {
  const component = renderer.create(
    <InfoText 
    color="green"
    children={  
      <p><span className="font-medium">Warning:</span> Warning text here</p> 
    }
  />
);
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});


it('should render the element with default (primary) colour and text', () => {
  const component = renderer.create(
    <InfoText 
    children="This is a test"
  />
);
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});


it('should render the element with red plus the html stated in children', () => {
  const component = renderer.create(
    <InfoText 
    color="red"
    children={
      <p><span className="font-medium">Warning:</span> Warning text here</p> 
    }
    />
);
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});