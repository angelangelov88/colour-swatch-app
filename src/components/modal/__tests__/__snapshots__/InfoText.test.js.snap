// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the element in green with heading with bold text and text 1`] = `
<div
  className="my-6 py-5 px-6 text-sm text-green-100 bg-green-10 shadow-[inset_6px_0px_0px_#46DBAA]"
>
  <p>
    <span
      className="font-medium"
    >
      Warning:
    </span>
     Warning text here
  </p>
</div>
`;

exports[`should render the element in yellow with no text 1`] = `
<div
  className="my-6 py-5 px-6 text-sm text-yellow-100 bg-yellow-10 shadow-[inset_6px_0px_0px_#F0B618]"
/>
`;

exports[`should render the element with default (primary) colour and text 1`] = `
<div
  className="my-6 py-5 px-6 text-sm text-primary-100 bg-primary-10 shadow-[inset_6px_0px_0px_#4456F6]"
>
  This is a test
</div>
`;

exports[`should render the element with red plus the html stated in children 1`] = `
<div
  className="my-6 py-5 px-6 text-sm text-red-100 bg-red-10 shadow-[inset_6px_0px_0px_#F64A44]"
>
  <p>
    <span
      className="font-medium"
    >
      Warning:
    </span>
     Warning text here
  </p>
</div>
`;

exports[`should render the element with red text and red background plus the text stated in children 1`] = `
<div
  className="my-6 py-5 px-6 text-sm text-red-100 bg-red-10 shadow-[inset_6px_0px_0px_#F64A44]"
>
  This is a test
</div>
`;
